payjoin-cli/src/app.rs

```diff
@@ -41,22 +45,97 @@ impl App {
         Ok(Self { config, bitcoind, seen_inputs })
     }
 
+    #[cfg(feature = "v2")]
+    pub async fn send_payjoin(&self, bip21: &str) -> Result<()> {
+        // TODO extract requests inside poll loop for unique OHTTP payloads
+        let req_ctx = self.create_pj_request(bip21)?;
+
+        let client = reqwest::Client::builder()
+            .danger_accept_invalid_certs(self.config.danger_accept_invalid_certs)
+            .build()
+            .with_context(|| "Failed to build reqwest http client")?;
+
+        log::debug!("Awaiting response");
+        let res = self.long_poll_post(&client, req_ctx).await?;
+        self.process_pj_response(res)?;
+        Ok(())
+    }
+
+    #[cfg(feature = "v2")]
+    async fn long_poll_post(
+        &self,
+        client: &reqwest::Client,
+        req_ctx: payjoin::send::RequestContext<'_>,
+    ) -> Result<Psbt> {
+        loop {
```

Comment:
thats a nice solution for this!
could we span a separate thread for this and make a single request(with timeout parameter set) and leave the main thread to return "awaiting.." ?

Reply:
HTTP requests time out after ~30 seconds, so we have to poll.

Other strategies mentioned in this PR had that sort of approach using asynchronous streaming protocols like WebSockets, but those protocols are succeptible to intersection timing attacks where the relay can connect IP addresses to participants.

One strategy to mitigate this kind of intersection attack without Tor or optional VPN is to use a mandatory OHTTP proxy in front of the payjoin relay, but to do that we either have to use WebTransport (new, heavy, but works inside HTTP) or HTTP polling. To keep this system simple enough that Bitcoin Core could implement it, after much discussion with devs, I landed on HTTP polling. 

More reasoning regarding the protocol design can be found [here on delving bitcoin](https://delvingbitcoin.org/t/serverless-payjoin-protocol-design/96)